tizen-api/src/tizen-api-util.c

```diff
@@ -0,0 +1,442 @@
+        ml_info->info[i].type = ML_TENSOR_TYPE_UINT8;
+        break;
+      case _NNS_FLOAT64:
+        ml_info->info[i].type = ML_TENSOR_TYPE_FLOAT64;
+        break;
+      case _NNS_FLOAT32:
+        ml_info->info[i].type = ML_TENSOR_TYPE_FLOAT32;
+        break;
+      case _NNS_INT64:
+        ml_info->info[i].type = ML_TENSOR_TYPE_INT64;
+        break;
+      case _NNS_UINT64:
+        ml_info->info[i].type = ML_TENSOR_TYPE_UINT64;
+        break;
+      default:
+        ml_info->info[i].type = ML_TENSOR_TYPE_UNKNOWN;
+        break;
+    }
+
+    /* Set dimension */
+    for (j = 0; j < max_dim; j++) {
+      ml_info->info[i].dimension[j] = gst_info->info[i].dimension[j];
+    }
+
+    for ( ; j < ML_TENSOR_RANK_LIMIT; j++) {
+      ml_info->info[i].dimension[j] = 1;
+    }
+  }
+}
+
+/**
+ * @brief Copies tensor meta info from gst tensots info.
+ */
+void
+ml_util_copy_tensors_info_from_ml (GstTensorsInfo * gst_info,
+  const ml_tensors_info_s * ml_info)
+{
+  guint i, j;
+  guint max_dim;
+
+  if (!gst_info || !ml_info)
+    return;
+
+  gst_tensors_info_init (gst_info);
+  max_dim = MIN (ML_TENSOR_RANK_LIMIT, NNS_TENSOR_RANK_LIMIT);
+
+  gst_info->num_tensors = ml_info->num_tensors;
+
+  for (i = 0; i < ml_info->num_tensors; i++) {
+    /* Copy name string */
+    if (ml_info->info[i].name) {
+      gst_info->info[i].name = g_strdup (ml_info->info[i].name);
+    }
+
+    /* Set tensor type */
+    switch (ml_info->info[i].type) {
+      case ML_TENSOR_TYPE_INT32:
+        gst_info->info[i].type = _NNS_INT32;
+        break;
+      case ML_TENSOR_TYPE_UINT32:
+        gst_info->info[i].type = _NNS_UINT32;
+        break;
+      case ML_TENSOR_TYPE_INT16:
+        gst_info->info[i].type = _NNS_INT16;
+        break;
+      case ML_TENSOR_TYPE_UINT16:
+        gst_info->info[i].type = _NNS_UINT16;
+        break;
+      case ML_TENSOR_TYPE_INT8:
+        gst_info->info[i].type = _NNS_INT8;
+        break;
+      case ML_TENSOR_TYPE_UINT8:
+        gst_info->info[i].type = _NNS_UINT8;
+        break;
+      case ML_TENSOR_TYPE_FLOAT64:
+        gst_info->info[i].type = _NNS_FLOAT64;
+        break;
+      case ML_TENSOR_TYPE_FLOAT32:
+        gst_info->info[i].type = _NNS_FLOAT32;
+        break;
+      case ML_TENSOR_TYPE_INT64:
+        gst_info->info[i].type = _NNS_INT64;
+        break;
+      case ML_TENSOR_TYPE_UINT64:
+        gst_info->info[i].type = _NNS_UINT64;
+        break;
+      default:
+        gst_info->info[i].type = _NNS_END;
+        break;
+    }
+
+    /* Set dimension */
+    for (j = 0; j < max_dim; j++) {
+      gst_info->info[i].dimension[j] = ml_info->info[i].dimension[j];
+    }
+
+    for ( ; j < NNS_TENSOR_RANK_LIMIT; j++) {
+      gst_info->info[i].dimension[j] = 1;
+    }
+  }
+}
+
+/**
+ * @brief Checks the availability of the given execution environments.
+ */
+int
+ml_util_check_nnfw (ml_nnfw_e nnfw, ml_nnfw_hw_e hw)
+{
+  /** @todo fill this function */
+#ifndef ENABLE_TENSORFLOW_LITE
```

Comment:
You cannot determine this at build time.
If someone installs tensorflow-lite subplugin (built separatedly with /ext/nnstreamer/...), you get this supported.
If someone deletes the subplugin, you cannot support it.

You may want to use ```get_subplugin``` API (nnstreamer_subplugin.h).

Reply:
Updated to check tensorflow using nnstreamer_filter_find()